### DISCUSSION TITLE
so docker containers do not start if DNS resoving/network problems exists at boot time

### DISCUSSION THREAD

**Discussion by maseron** [Security-Onion-Solutions/securityonion#7214] [Unsupported Versions] [answered]
Hello!

I noticed that so docker containers does not start if at boot time there is no working DNS or there is network problems to dns server. And looking deeper it get stuck on `yum check-update 2>&1 | grep "Error: rpmdb open failed` what is in file /opt/so/saltstack/default/salt/repo/client/init.sls

I did quick test with adding  fake repo.securityonion.net record to hosts file `echo "127.0.0.1 repo.securityonion.net" >> /etc/hosts`, rebooted the securityonion and after that the security onion docker containers started without problem.

Onion version: 2.3.91, installed from ISO as standalone.
Will test if the same thing happens with latest version 2.3.100, but looking on the init.sls change history it looks like the result will be the same.

BR,
Martins

**Comment by dougburks** [comment, marked as answer]
If you expect your DNS resolution to be inconsistent, then you may want to consider selecting the `airgap` option so that your installation is less dependent on external resources.

**Comment by maseron** [comment]
But will it not make harder to update the system? For me it looks littlebit incorrect that unsuccessful update check restricts the system startup. For me it's easier to add/remove the one record in /etc/hosts file than trying to understand what should be changed in airgap system to get updates. 
It's not like usual situation that the DNS is not working. So far I have experienced this only 2 times.

**Reply by dougburks** [reply]
Use whatever works best for you.
